SERIAL_7E1 & Correctif pour ESP32 et Arduino Leonardo

iodeo opened the pull request:
Voir les descriptions des commits pour plus de détails

Cette version a été testé avec esp32_wroom32e / esp32_wrover / arduino_leonardo(atmega32u4)

Resumé des modifs :

- Utilisation de la config SERIAL_7E1 de hardwareSerial (calcul de parité natif des noyaux arduino et esp32)
- On ne ferme pas la liaison série au moment de changer la vitesse pour l'esp32 car cela parasite l'acquittement dans certains cas. La fermeture reste nécessaire pour les arduinos (ajout macro #else)
- Renommage de la variable SPEED en _SPEED (conflit avec le noyau)
- Ajout d'un constructeur pour ESP32 permettant la relocalisation des broches à utiliser pour le port Série. Les paramètres rxPin et txPin sont relatifs à l'esp32 : rxPin va au tx du minitel et txPin va au rx du minitel

iodeo committed:
use of SERIAL_7E1 && unuse .end for esp32

iodeo committed:
modify comment for esp32 serial flush

iodeo committed:
add constructor for esp32 to enable serial pin mapping

iodeo committed:
add constructor for ESP32 to enable serial pin mapping - cpp

iodeo committed:
Correctif changeSpeed pour esp32

Le changement de vitesse avec begin(bauds) est trop long avec l'esp32. L'acquittement du minitel est parfois mal réceptionné.

L'utilisation de la fonction updateBaudRate(bauds) permet de régler ce problème

64rulez commented:
... Je découvre 'vite fait' ta modif (7E1). Il y a quelque temps, j'ai un peu bricolé cette librairie (quelques trucs qui ne me convenaient qu'à moitié, en particulier sur les acquitements, mais pas que .... des ajouts aussi (support du DRCS)).

Il faudrait qu'on en discute à l'occasion : je pense à faire aussi des changements pour support du 8N1 et le mode photo.

[Aussi, des modifs dans Minitel_ESP32]

En attendant, un avant goût de mes travaux actuels sur M2 (avec Minitel_ESP32 + Minitel1B_Hard modifiés)
==> Encore un peu de travail sur l'ampli vidéo (aussi sur la qualité de l'acquisition/conversion)

Test of RGB2HDMI with TS9347_40 profile (Alt2)
Overall view, 95% operationnal

Overall https://youtu.be/2ILQbh81fa4
80 https://youtu.be/him87SG9PTo
BW https://youtu.be/GoLXB4IvrKo
FG https://youtu.be/INuxbzYLp60
x1 https://youtu.be/XcGI95jKbYg
x8 https://youtu.be/I7CZI2Axcuk

iodeo commented:
Si les DRCS et 8N1 sont spécifiques au minitel2, peut être faut-il envisager une lib Minitel2 ?
Je serais d'avis de garder celle-ci le plus fidèle possible à la STUM1B, mais c'est à @eserandour de voir ça ;)

iodeo committed:
Add timeout in identificationBytes function

Some minitels don't answer to identifyDevice request
Timeout is needed to avoid being stucked in while loop

iodeo commented:
Salut,
Je sais pas si tu auras le temps d'intégrer les modifs mais en voici une dernière concernant la fonction identifyDevice.
J'ai un minitel qui ne répond pas à la requête, le programme s'enferme dans une boucle while en attendant la réponse. J'ai ajouté un timeout.
Au pire le code modifié est dispo sur mon fork

sblendorio committed:
update

sblendorio committed:
update

sblendorio committed:
update

sblendorio committed:
update

iodeo committed:
Merge pull request #1 from sblendorio/master

Improved compatibility with Minitel1 (and 1B)

iodeo committed:
searchSpeed: index incrementation corrected

out of bound _SPEED[4] was reached due to post-expression incrementation
